projects/onto-search/src/lib/onto-search-column-selector/onto-search-column-selector.component.ts

```diff
@@ -0,0 +1,80 @@
+import {Component, EventEmitter, Input, OnChanges, Output, SimpleChanges} from '@angular/core';
+import {MatSelectChange} from '@angular/material/select';
+import {FormControl} from '@angular/forms';
+import {SelectionColumn} from './models/selection-column';
+import {SelectionColumnGroup} from './models/selection-column-group';
+
+@Component({
+  selector: 'onto-search-column-selector',
+  templateUrl: './onto-search-column-selector.component.html',
+  styleUrls: ['./onto-search-column-selector.component.scss']
+})
+export class OntoSearchColumnSelector implements OnChanges {
+  selection: SelectionColumn[] = [];
+  selectCtrl = new FormControl();
+
+  /**
+   * The groups of columns information for the dropdown select
+   */
+  @Input()
+  columnGroups: SelectionColumnGroup[];
+
+  /**
+   * Sets selected columns in dropdown select
+   */
+  @Input()
+  selectedColumns: SelectionColumn[];
+
+  /**
+   * Sets default columns. Shown when no selectedColumns are passed and when
+   * reset button is pressed
+   */
+  @Input()
+  defaultColumns: SelectionColumn[];
+
+  /**
+   * Fired when the dropdown panel is closed.
+   * Sends an array {@link SelectionColumn}
+   */
+  @Output()
+  selectionChanged: EventEmitter<SelectionColumn[]> = new EventEmitter<SelectionColumn[]>();
+
+  constructor() {
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    const defaultColumns = changes?.defaultColumns?.currentValue;
+    const selectedColumns = changes?.selectedColumns?.currentValue;
+    if (defaultColumns) {
```

Comment:
Don't you need to check if there are some default values actually? As the defaultColumns is an array, if you check for the reference only, you'll get true even if it's empty